File: (8 kyu) Return Negative/(8 kyu) Return Negative.js
// #1
// function makeNegative(num) {
//   if (num < 0) {
//     return num;
//   } else {
//     return (num = num * -1);
//   }
// }
// #2
// function makeNegative(num) {
//   if (num < 0) {
//     return num;
//   } else {
//     return -num;
//   }
// }
// #3
// function makeNegative(num) {
//   return num < 0 ? num : -num;
// }
// #4
// const makeNegative = (num) => (num < 0 ? num : -num);
// #5
const makeNegative = (num) => -Math.abs(num);

File: 8kyu/(8 kyu) Return Negative/(8 kyu) Return Negative.dart
// 1 Simple solution
num makeNegative(n) {
  if (n > 0) {
    return -n;
  } else {
    return n;
  }
}

// 2 Optimized solution
num makeNegative(n) {
  return n > 0 ? -n : n;
}

// 3 Clever solution
num makeNegative(n) {
  return -n.abs();
}

// 4 Coding golf
num makeNegative(n) => n > 0 ? -n : n;
